backend/database-fs.js

```diff
@@ -7,9 +7,39 @@ import { checkKey } from '~/shared/domains/chelonia/db.js'
 // Initialized in `initStorage()`.
 let dataFolder = ''
 
+// Some operating systems (such as macOS and Windows) use case-insensitive
+// filesystems by default. This can be problematic for Chelonia / Group Income,
+// as we rely on keys being case-sensitive. This is especially relevant for CIDs,
+// where collissions could lead to DoS or data corruption.
+async function testCaseSensitivity () {
+  const date = new Date()
+  const dateString = date.toISOString()
+  const originalKey = `_private_testCaseSensitivity_${date.getTime()}_${(0, Math.random)().toFixed(8).slice(2)}`
+  // We only replace the first `p` with `P`. We don't use, e.g., `.toUpperCase()`
+  // because case conversion depends on the locale, and `i` may sometimes be
+  // `I` and sometimes be `İ`.
+  const differentlyCasedKey = '_P' + originalKey.slice(2)
+  await writeData(originalKey, dateString)
+  try {
+    const valueOriginalCase = await readData(originalKey)
+    const valueDifferentCase = await readData(differentlyCasedKey)
+    if (valueOriginalCase?.toString() !== dateString) {
+      console.error(`Unexpected value on case-sensitivity test; expected ${dateString}`)
+      throw new Error('Unexpected value: original key does not have the correct value')
+    }
+    if (valueDifferentCase?.toString() === dateString) {
+      console.error(`Unexpected value on case-sensitivity test; expected a differently-cased key to have a value other than ${dateString}`)
+      throw new Error('Unexpected value: key with different case has the same value as the origical key')
```

Comment:
Could you please update these error messagse (for both the console.error and the exception thrown) to:

> Filesystem database backend only works on case-sensitive filesystems. This appears to be a case insensitive file system.